editor/src/components/user-bar.tsx

```diff
@@ -0,0 +1,163 @@
+import { ClientSideSuspense } from '@liveblocks/react'
+import React from 'react'
+import { useOthers, useSelf, useStatus } from '../../liveblocks.config'
+import { getUserPicture, isLoggedIn } from '../common/user'
+import type { MultiplayerColor } from '../core/shared/multiplayer'
+import {
+  multiplayerColorFromIndex,
+  multiplayerInitialsFromName,
+  normalizeMultiplayerName,
+  normalizeOthersList,
+} from '../core/shared/multiplayer'
+import { Avatar, Tooltip, useColorTheme } from '../uuiui'
+import { Substores, useEditorState } from './editor/store/store-hook'
+
+const MAX_VISIBLE_OTHER_PLAYERS = 4
+
+export const UserBar = React.memo(() => {
+  const loginState = useEditorState(
+    Substores.userState,
+    (store) => store.userState.loginState,
+    'UserBar loginState',
+  )
+
+  const roomStatus = useStatus()
+
+  if (!isLoggedIn(loginState)) {
+    return null
+  } else if (roomStatus !== 'connected') {
+    return <SinglePlayerUserBar />
+  } else {
+    return (
+      <ClientSideSuspense fallback={<div />}>{() => <MultiplayerUserBar />}</ClientSideSuspense>
+    )
+  }
+})
+UserBar.displayName = 'UserBar'
+
+export const SinglePlayerUserBar = React.memo(() => {
+  const userPicture = useEditorState(
+    Substores.userState,
+    (store) => getUserPicture(store.userState.loginState),
+    'SinglePlayerUserBar userPicture',
+  )
+  return (
+    <a href='/projects' target='_blank'>
+      <Avatar userPicture={userPicture} isLoggedIn={true} />
+    </a>
+  )
+})
+SinglePlayerUserBar.displayName = 'SinglePlayerUserBar'
+
+const MultiplayerUserBar = React.memo(() => {
+  const colorTheme = useColorTheme()
+
+  const self = useSelf()
+  const myName = React.useMemo(
```

Comment:
I suspect you can get away with not needing to wrap this in a `useMemo`, as it looks to be a string and I'm guessing the cost of `normalizeMultiplayerName` isn't that high?